### DISCUSSION TITLE
Ispindel does not wake up after WiFi was lost once

### DISCUSSION THREAD

**Discussion by ppet1987** [universam1/iSpindel#550] [Q&A]
Hello, 
I am not sure if this is the correct place to post my issue with the ISpindel. 

I built an ISpindel on a developer board and flashed it with the latest firmware (7.1.0). I am using ubidots for data handling. 
It is working quite nicely, as long as it has WiFi connection. 
However, I came across an interesting behavior: 
If I start the Spindel with measuring in a 900s interval, it does its job. If it wakes up without having the chance to connect to WLAN (router off) for this specific datapoint it will no longer wake up anymore. It rather falls into deep sleep forever. 

Has anyone have any advice on this?
I expected the Spindel to simply wake up after the next 900s and if Wlan is available, it uploads the next datapoint, however it does not in my case. 
I know that I have to improve my WLAN connectivity in the cellar (where my beer ferments). Nevertheless, I think that loosing one datapoint due to connectivity must not crash the spindel.
Thank-you in advance!
BR, 
Patrik

**Comment by ErikdBr** [comment]
> I expected the Spindel to simply wake up after the next 900s and if Wlan is available, it uploads the next datapoint

Yes it should do that. Are you sure you don't have a deep sleep issue due to use of a resistor instead of a bat43 diode?

**Reply by ppet1987** [reply]
Hi Erik, 

that's a good point. However, I built the Spindel with a bat43. So I excluded the deep sleep theory. 
Furthermore, I have the feeling, that the battery is also drained faster, once it runs into the "unable-to-connect" problem. 
I think I will have to recreate the error with a shorter data interval and additional Putty listening...
Maybe this helps. 

BR

**Comment by ppet1987** [comment]
An update, as I did some error seeking:

I changed to firmware 1.6.5, as it seemed to be the most stable one, currently (according to Hobbybrauer forum aso). 

However, I experienced the same behavior there. 
So I reduced logging time to 30s and moved within and out of WiFi range to force my Ispindel into the error. 
It seemed that it is not really a matter of wether it could connect in the workers run before. 
What I could trace is this: 

![error](https://user-images.githubusercontent.com/92568925/137617028-6cd17791-ac59-445c-a871-7e1d8d36b1b0.png)

It seems that the Wemo gets stuck during wakeup. 
Any advice on this? I would assume replacing is the only option?

Thanks in advance!
BR, 
Patrik

**Reply by mp-se** [reply]
Check the bat diode for the corret orientarion and soldering or replace the diode first.

**Reply by ppet1987** [reply]
I checked the diode and reheated the solder points. However, no change. So I replaced the Wemos and now it’s running stable. Seems like the first Wemos has an issue when it comes to wake up. 
Br, 
Patrik

**Reply by pppedrillo** [reply]
@ppet1987 Funny json in the console log :) 
So you solved your issue? Was it dodgy WemOS module?

**Comment by thegreatgunbantoad** [comment]
A lot of the Wemos boards are clones and the quality can be quite variable.

**Comment by Ziglar1** [comment]
Same here, when powering on it updates the values (at brewfather). after that no single wake-up after the set time, any ideas ?

**Reply by pppedrillo** [reply]
Yes. Use search with closed issues included  - discussed hundred of times.
